src/pudl/scripts/dbt_helper.py

```diff
@@ -325,19 +311,58 @@ def _write_row_counts(row_counts: pd.DataFrame, target: str = "etl-full"):
 
 def update_row_counts(
     table_name: str,
-    partition_column: str = "report_year",
+    data_source: str,
     target: str = "etl-full",
     clobber: bool = False,
     update: bool = False,
 ) -> UpdateResult:
     """Generate updated row counts per partition and write to csv file within dbt project."""
+    model_path = _get_model_path(table_name, data_source)
+    schema_path = model_path / "schema.yml"
+    schema = DbtSchema.from_yaml(schema_path)
+
+    partition_columns = _extract_row_count_partitions(schema)
```

Comment:
`_extract_row_count_partitions()` expectes a `DbtTable` rather than a `DbtSchema` so this call fails -- the `DbtSchema` has no `data_tests` attribute from which the partition column can be extracted.

Reply:
Reversed the cleanup and will tackle that in a separate PR that includes the other issues you mentioned